src/parser.jl

```diff
@@ -32,9 +32,17 @@ function function_body_lines!(flines, ast::Expr, infunction)
     else
         args = ast.args
     end
-    infunction |= isfuncexpr(ast)
-    for arg in args
-        flines = function_body_lines!(flines, arg, infunction)
+
+    if isfuncexpr(ast) && length(args)>=2 && args[2] isa Expr
```

Comment:
Could you perhaps add a comment that explains what's going on? (No worries, I won't hold this PR hostage if you don't have time to do that, given that the rest of the code is not exactly a shining beacon for well-commented code to start with... but since you clearly understand what you are doing, it might not be too hard for you to write something, and it'd help others who are trying to figure out what this code does in the future...

E.g. people like me... I'll now go to read up on this code to figure out what the "length(args)>=2 && args[2] isa Expr" bit is about and then come back to conduct a proper review ;-).

Reply:
OK, looking at https://docs.julialang.org/en/v1/devdocs/ast/ it says:
> A basic function definition is parsed as (function (call f x) body). 

So I guess the code above is meant to check if there is a function body, and in that case, skip the arguments?
But is there actually a case when length(args) is different from 2?  Hm, perhaps in that weird second type of ast expressions for which `isfuncexpr` returns true? Can somebody explain that bit of `isfuncexpr` to me? I.e. the bit after `ex.head == :function || ` in the following definition:
```julia
isfuncexpr(ex::Expr) =
    ex.head == :function || (ex.head == :(=) && typeof(ex.args[1]) == Expr && ex.args[1].head == :call)
```

Reply:
@fingolfin I've added a whole bunch of comments. If something is still unclear, let me know!